### PR TITLE
Update extensions doc

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: docs/source/extensions/callbacks.rst [hunk 1]
```diff
@@ -60,8 +60,8 @@ Example:
 
 .. testoutput::
 
-    Starting to init trainer!
-    trainer is init now
+    Starting to initialize trainer!
```
Comment (carmocca) [inline docs/source/extensions/callbacks.rst:63]:
```suggestion
    Starting to initialize the trainer!
```

File: docs/source/extensions/callbacks.rst [hunk 2]
```diff
@@ -47,10 +47,10 @@ Example:
 
     class MyPrintingCallback(Callback):
         def on_init_start(self, trainer):
-            print("Starting to init trainer!")
+            print("Starting to initialize trainer!")
```
Comment (carmocca) [inline docs/source/extensions/callbacks.rst:50]:
```suggestion
            print("Starting to initialize the trainer!")
```

File: docs/source/extensions/loops.rst [hunk 3]
```diff
@@ -222,7 +221,7 @@ Loop API
 --------
 Here is the full API of methods available in the Loop base class.
 
-The :class:`~pytorch_lightning.loops.base.Loop` class is the base for all loops in Lighting just like the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base for all models.
```
Comment (carmocca) [inline docs/source/extensions/loops.rst:225]: Might be worth grepping "Lighting" and fixing all typos at once 😂

File: pytorch_lightning/callbacks/base.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def state_key(self) -> str:
         """Identifier for the state of the callback.
 
         Used to store and retrieve a callback's state from the checkpoint dictionary by
-        ``checkpoint["callbacks"][state_key]``. Implementations of a callback need to provide a unique state key if 1)
+        ``checkpoint["callbacks"]["state_key"]``. Implementations of a callback need to provide a unique state key if 1)
```
Comment (carmocca) [inline pytorch_lightning/callbacks/base.py:41]: I think this was referring to a variable

**Review by carmocca — APPROVED** [review]

File: docs/source/extensions/callbacks.rst [hunk 5]
```diff
@@ -197,245 +198,310 @@ The following are best practices when using/designing callbacks.
...
+
+state_key
+~~~~~~~~~
+
+.. autoattribute:: pytorch_lightning.callbacks.Callback.state_key
+    :noindex:
+
+
+Hooks
+^^^^^^^^^^
```
Comment (Borda) [inline docs/source/extensions/callbacks.rst:219]:
```suggestion
^^^^^
```

**Review by Borda — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]
❤️

File: pytorch_lightning/callbacks/base.py (hunk 4 above)
Comment (rohitgr7) [reply pytorch_lightning/callbacks/base.py:41]: done!

File: docs/source/extensions/loops.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/extensions/loops.rst:225]: haha... done!!

File: docs/source/extensions/loops.rst [hunk 6]
```diff
@@ -50,7 +50,7 @@ Think of this as swapping out the engine in a car!
 
 ----------
 
-Understanding the default Trainer loop
+Understanding the default Trainer Loop
```
Comment (s-rog) [inline docs/source/extensions/loops.rst:53]:
```suggestion
Understanding the Default Trainer Loop
```

File: docs/source/extensions/loops.rst [hunk 7]
```diff
@@ -109,12 +109,12 @@ Defining a loop within a class interface instead of hard-coding a raw Python for
...
 .. _override default loops:
 
-Overriding the default loops
+Overriding the default Loops
 ----------------------------
 
 The fastest way to get started with loops, is to override functionality of an existing loop.
-Lightning has 4 main loops it uses: :class:`~pytorch_lightning.loops.fit_loop.FitLoop` for training and validating,
-:class:`~pytorch_lightning.loops.dataloader.evaluation_loop.EvaluationLoop` for testing,
+Lightning has 4 main loops it uses: :class:`~pytorch_lightning.loops.fit_loop.FitLoop` for fitting (training and validating),
```
Comment (tchaton) [inline docs/source/extensions/loops.rst:116]:
```suggestion
Lightning has 4 main loops which relies on : :class:`~pytorch_lightning.loops.fit_loop.FitLoop` for fitting (training and validating),
```

File: docs/source/extensions/loops.rst [hunk 8]
```diff
@@ -212,8 +212,7 @@ Finally, attach it into the :class:`~pytorch_lightning.trainer.trainer.Trainer`:
     # fit() now uses your fancy loop!
     trainer.fit(...)
 
-Now you have full control over the Trainer.
-But beware: The power of loop customization comes with great responsibility.
+But beware: Loop customization gives you more power and full control over the Trainer and with great power comes great responsibility.
```
Comment (tchaton) [inline docs/source/extensions/loops.rst:215]: Neat !

File: docs/source/extensions/loops.rst [hunk 9]
```diff
@@ -222,7 +221,7 @@ Loop API
 --------
 Here is the full API of methods available in the Loop base class.
 
-The :class:`~pytorch_lightning.loops.base.Loop` class is the base for all loops in Lighting just like the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base for all models.
+The :class:`~pytorch_lightning.loops.base.Loop` class is the base for all loops in Lightning just like the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base for all models.
```
Comment (tchaton) [inline docs/source/extensions/loops.rst:224]:
```suggestion
The :class:`~pytorch_lightning.loops.base.Loop` class is the base of all loops in the same way as the :class:`~pytorch_lightning.core.lightning.LightningModule` is the base of all models.
```

File: docs/source/extensions/loops.rst [hunk 10]
```diff
@@ -390,7 +396,8 @@ To run the following demo, install Flash and `BaaL <https://github.com/ElementAI
         torch.nn.Dropout(p=0.1),
         torch.nn.Linear(512, datamodule.num_classes),
     )
-    model = ImageClassifier(backbone="resnet18", head=head, num_classes=datamodule.num_classes, serializer=Probabilities())
+    model = ImageClassifier(backbone="resnet18", head=head, num_classes=datamodule.num_classes, output=Probabilities())
```
Comment (tchaton) [inline docs/source/extensions/loops.rst:399]: Neat !

**Review by tchaton — APPROVED** [review]
LGTM!
